docs/write_test/concepts.md

```diff
@@ -88,6 +90,35 @@ example, when defining a notifier, you can define its configuration structure
 together. The corresponding configuration will be loaded into the notifier with
 the same schema. The extended configuration also supports variables.
 
+## Combinator
+
+The combinator helps to run large-scale test cases with different variables,
+such as running multiple images with different VM sizes or other variables.
+There are two kinds of combinators provided: grid combinator and batch
+combinator. You can also write your own combinator.
+
+### Grid combinator
+
+A [grid combinator](../../lisa/combinators/grid_combinator.py) selects all the
+combination of provided images and variables. For example, if you are to run 10
+images with 3 different VM sizes ([`img1`, `vm1`], [`img1`, `vm2`], [`img1`,
+`vm3`], [`img2`, `vm1`], etc.), the grid combinator saves you from expanding
+that combination out manually in complete form and would automatically give all
+$10\times3=30$ combinations.
+
+### Batch combinator
+
+A [batch combinator](../../lisa/combinators/batch_combinator.py) instead runs a
+batch of specified combinations. For example, maybe we only want `img1` to run
+with `vm1` but we want `img2` to run with `vm1`, `vm2` and `vm3`, you can give
+the batch combinator such specification and all tests would run as you expect.
+
+## Transformer
+
+The transformers generate variables from other variables, and multiple
+transformers can run one by one to achieve complex transformation. For example,
+one transformer can build kernel and another one can save it to a VHD.
```

Comment:
What does "build kernel" and "save to VHD" mean here?

Reply:
Build linux kernel from source code, and save the VM a VHD. They can be two steps, implemented by separated transformers. The steps can be composited in other workflows.
I will add some words to make it clearer.

Reply:
I did not see anything even close to something like issuing a kernel build coming from this Transformer thing, that's why I'm super confused...

Reply:
LISAv2 has that capability, and it's one of the guest test scenario. It supports to test backport kernels, or a patch.